core/storage/suggestion/gae_models.py

```diff
@@ -536,6 +582,44 @@ def get_in_review_question_suggestions(
             cls.author_id != user_id
         )).fetch(feconf.DEFAULT_SUGGESTION_QUERY_LIMIT)
 
+    @classmethod
+    def get_in_review_question_suggestions_by_offset(
+        cls, limit: int, offset: int, user_id: str
+    ) -> Tuple[Sequence[GeneralSuggestionModel], int]:
+        """Fetches question suggestions that are in-review and not authored by
+        the supplied user.
+
+        Args:
+            limit: int. Maximum number of entities to be returned.
+            offset: int. Number of query results to skip.
+            user_id: str. The id of the user trying to make this query. As a
+                user cannot review their own suggestions, suggestions authored
+                by the user will be excluded.
+
+        Returns:
+            Tuple of (results, next_offset). Where:
+                results: list(SuggestionModel). A list of suggestions that are
+                    in-review, not authored by the supplied user, and that match
+                    one of the supplied language codes.
+                next_offset: int. Number of query results to skip on a
+                    subsequent fetch call.
+        """
+        suggestion_query = cls.get_all().filter(datastore_services.all_of(
+            cls.status == STATUS_IN_REVIEW,
+            cls.suggestion_type == feconf.SUGGESTION_TYPE_ADD_QUESTION,
+            cls.author_id != user_id
+        ))
+
+        results: Sequence[GeneralSuggestionModel] = (
+            suggestion_query.fetch(limit, offset=offset)
+        )
+        next_offset = offset + len(results)
+
+        return (
+            results,
+            next_offset
```

Comment:
Just a question -- if we reach the end of the list, what is the behaviour here? Will the subsequent query with next_offset just not do anything? That could be a bit weird for the user.

I think what we've done in such cases is to fetch N + 1 results and return N, but use the info that if the result list for the "N+1" query has <= N results then we've reached the end of the list. (Though I'm not sure which value of "next_offset" is supposed to represent "the end of the list".)

Ditto below.

(EDIT: Just read the frontend code, and it looks like some of the offsetting is being handled there as well. I think that seems sensible, but perhaps it might be worth additionally doing the above so that the backend can pass a clear signal to the frontend too? It seems reasonable to me to have two separate things here because the frontend is independently maintaining its own cached results and display logic.)

Reply:
I think that's reasonable. But wouldn't you still need to fetch N+1 within this function to conclude with certainty that this is the last batch?

Reply:
I believe we decided to handle the "more" functionality in the frontend (https://github.com/oppia/oppia/issues/13462). This way, it's the frontend's responsibility to "fetch ahead" to figure out whether there are more results and the backend storage functions can remain simple. 

Perhaps @vojtechjelinek can provide additional context as well.